### ISSUE TITLE
Code license?

### ISSUE THREAD

**Issue by exelotl** [zevv/nimcoro#1]
Hey zevv - I'm looking to release a project that uses some code based on nimcoro, was wondering if you'd be ok to add an open-source license to this repo? Thanks :)

**Comment by zevv** [comment]
sure, MIT `LICENSE` was added. Have fun!

**Comment by zevv** [comment]
(If you want to boldly  go where no one has gone before, use CPS instead! https://github.com/zevv/cpstest)

**Comment by exelotl** [comment]
Thanks a bunch! I'm actually really happy with this solution over CPS or closure iterators for my project, as I tweaked nimcoro to avoid `ref` objects, and allow the stack to be passed in, making the whole thing possible with zero heap allocation. Works great with `--gc:none` (which was essential back when I started the project, but still has some desirable qualities even now). It's also super unobtrusive - clean stack traces in GDB, no pragmas/annotations, no hidden gotchas (for-loops etc.). 

That said, CPS looks brilliant and I'm looking forward to trying it in the future :)
